comp/otelcol/status/statusimpl/status.go

```diff
@@ -0,0 +1,248 @@
+	"embed"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io"
+	"net/http"
+	"sync"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/status"
+	ddflareextension "github.com/DataDog/datadog-agent/comp/otelcol/ddflareextension/def"
+	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/prometheus"
+)
+
+// httpClients should be reused instead of created as needed. They keep cached TCP connections
+// that may leak otherwise
+var (
+	httpClient     *http.Client
+	clientInitOnce sync.Once
+)
+
+//go:embed status_templates
+var templatesFS embed.FS
+
+type dependencies struct {
+	fx.In
+
+	Config config.Component
+}
+
+type provides struct {
+	fx.Out
+
+	StatusProvider status.InformationProvider
+}
+
+type statusProvider struct {
+	Config         config.Component
+	receiverStatus map[string]interface{}
+	exporterStatus map[string]interface{}
+}
+
+type prometheusRuntimeConfig struct {
+	Service struct {
+		Telemetry struct {
+			Metrics struct {
+				Readers []struct {
+					Pull struct {
+						Exporter struct {
+							Prometheus struct {
+								Host string
+								Port int
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func client() *http.Client {
+	clientInitOnce.Do(func() {
+		httpClient = apiutil.GetClient(false)
+	})
+
+	return httpClient
+}
+
+// Module defines the fx options for the status component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newStatus))
+}
+
+func newStatus(deps dependencies) provides {
+	return provides{
+		StatusProvider: status.NewInformationProvider(statusProvider{
+			Config: deps.Config,
+			receiverStatus: map[string]interface{}{
+				"spans":           0.0,
+				"metrics":         0.0,
+				"logs":            0.0,
+				"refused_spans":   0.0,
+				"refused_metrics": 0.0,
+				"refused_logs":    0.0,
+			},
+			exporterStatus: map[string]interface{}{
+				"spans":          0.0,
+				"metrics":        0.0,
+				"logs":           0.0,
+				"failed_spans":   0.0,
+				"failed_metrics": 0.0,
+				"failed_logs":    0.0,
+			},
+		}),
+	}
+}
+
+// Name returns the name
+func (s statusProvider) Name() string {
+	return "OTel Agent"
+}
+
+// Section return the section
+func (s statusProvider) Section() string {
+	return "OTel Agent"
+}
+
+func (s statusProvider) getStatusInfo() map[string]interface{} {
+	statusInfo := make(map[string]interface{})
+
+	values := s.populateStatus()
+
+	statusInfo["otelAgent"] = values
+
+	return statusInfo
+}
+
+func getPrometheusURL(extensionResp ddflareextension.Response) (string, error) {
+	var runtimeConfig prometheusRuntimeConfig
+	if err := yaml.Unmarshal([]byte(extensionResp.RuntimeConfig), &runtimeConfig); err != nil {
+		return "", err
+	}
+	prometheusHost := "localhost"
+	prometheusPort := 8888
+	for _, reader := range runtimeConfig.Service.Telemetry.Metrics.Readers {
```

Comment:
This means if there are multiple readers, the last one will be used as the endpoint. Not sure if that's correct but it should be rare

Reply:
Good callout, I tested this with multiple readers and I believe `Readers` will only contain readers that follow the `pull.exporter.prometheus` pattern because of the `prometheusRuntimeConfig` struct.